src/renderer/src/components/song/context-menu/items/PlaylistChooser.tsx

```diff
@@ -1,91 +1,77 @@
+import SongImage from "../../SongImage";
 import DropdownList from "@renderer/components/dropdown-list/DropdownList";
-import { addNotice } from "@renderer/components/notice/NoticeContainer";
 import {
-  noticeError,
-  PLAYLIST_SCENE_LIST,
-  setCreatePlaylistBoxSong,
-  setPlaylistActiveScene,
-  setShowPlaylistCreateBox,
+  addToPlaylist,
+  deleteSongFromPlaylist,
+  setPlaylistActivePage,
 } from "@renderer/components/playlist/playlist.utils";
-import { setSidebarActiveTab, SIDEBAR_PAGES } from "@renderer/scenes/main-scene/main.utils";
-import { CheckIcon, CircleCheckIcon, PlusIcon } from "lucide-solid";
-import { Accessor, Component, For, Setter, Show } from "solid-js";
-import { Song } from "src/@types";
+import { setSidebarActiveTab } from "@renderer/scenes/main-scene/main.utils";
+import { SIDEBAR_PAGES } from "@renderer/scenes/main-scene/main.utils";
+import { CheckSquare, PlusIcon, Square } from "lucide-solid";
+import { Component, For, Show } from "solid-js";
+import { PlaylistDropdown, Song } from "src/@types";
 
 type PlaylistChooserProps = {
   song: Song;
-  playlistNames: string[];
-  setShowChooser: Setter<boolean>;
-  timeoutId: Accessor<NodeJS.Timeout | undefined>;
-  setTimeoutId: Setter<NodeJS.Timeout | undefined>;
+  playlists: PlaylistDropdown[];
+  onCreatePlaylistClick?: () => void;
 };
 
 const PlaylistChooser: Component<PlaylistChooserProps> = (props) => {
-  const addToPlaylist = async (name: string) => {
-    const result = await window.api.request("playlist::add", name, props.song);
-    if (result.isError) {
-      noticeError(result.error);
-      return;
-    }
-    addNotice({
-      title: "Song added",
-      description: "Successfully added song to playlist " + name + "!",
-      variant: "success",
-      icon: <CircleCheckIcon size={20} />,
-    });
-  };
-
-  const isInPlaylist = (song: Song, playlistName: string) => {
-    if (song.playlists === undefined) {
-      return false;
-    }
-
-    return song.playlists.includes(playlistName);
-  };
-
   return (
-    <DropdownList
-      onMouseOver={() => {
-        clearTimeout(props.timeoutId());
-      }}
-      onMouseLeave={() => {
-        props.setTimeoutId(
-          setTimeout(() => {
-            props.setShowChooser(false);
-          }, 320),
-        );
-      }}
-      class="max-h-72 w-40 overflow-auto pr-1.5"
-    >
+    <div class="flex flex-col gap-1">
```

Comment:
If you have a lot of playlists (or a small window) this doesn't scroll, initially i had fixed it by adding `h-screen overflow-y-scroll`, but i noticed that at the top (on the title bar) you can't click the items in the popover, so i used `h-72 overflow-y-scroll`.

Reply:
Added the scroll back.